src/OpenLoco/src/Map/MapGenerator.cpp

```diff
@@ -136,15 +142,261 @@ namespace OpenLoco::World::MapGenerator
         }
     };
 
+    enum class TopographyFlags : uint8_t
+    {
+        none = 0U,
+        hasHills = 1U << 0,
+        hasMountains = 1U << 1,
+    };
+    OPENLOCO_ENABLE_ENUM_OPERATORS(TopographyFlags);
+
+    // 0x004FD332
+    static constexpr std::array<TopographyFlags, 10> kTopographyStyleFlags = {
+        TopographyFlags::none,
+        TopographyFlags::none,
+        TopographyFlags::hasHills,
+        TopographyFlags::hasHills,
+        TopographyFlags::hasMountains | TopographyFlags::hasHills,
+        TopographyFlags::hasMountains | TopographyFlags::hasHills,
+        TopographyFlags::hasHills,
+        TopographyFlags::hasMountains | TopographyFlags::hasHills,
+        TopographyFlags::none,
+        TopographyFlags::hasMountains | TopographyFlags::hasHills,
+    };
+
     class OriginalTerrainGenerator
     {
+    private:
+        // 0x004626B7
+        void blitHill(const S5::Options& options, HeightMap& heightMap)
+        {
+            const uint32_t randomVal = getGameState().rng.randNext();
+            const bool flipHillImageLateral = randomVal & 0x100;
+            const bool flipHillImageVertical = randomVal & 0x200;
+
+            const auto hillShapesObj = ObjectManager::get<HillShapesObject>();
+            const uint8_t hillShapeCount = hillShapesObj->hillHeightMapCount + hillShapesObj->mountainHeightMapCount;
+
+            // The hill index calculation is a minor simpliciation compared to vanilla
+            const uint8_t randomHillIndex = (randomVal & 0xFF) % hillShapeCount;
+
+            const auto hillImage = hillShapesObj->image + randomHillIndex;
+            const auto g1Element = Gfx::getG1Element(hillImage);
+
+            auto featureWidth = g1Element->width & 0xFF;
+            auto featureHeight = g1Element->height & 0xFF;
+            if (flipHillImageLateral)
+            {
+                std::swap(featureWidth, featureHeight);
+            }
+
+            // 0x00462718
+            const auto randX = (randomVal >> 14) & 0x1FF;
+            const auto randY = (randomVal >> 23) & 0x1FF;
+            if (randX < 2)
+            {
+                printf("Bail! x = %d < 2\n", randX);
+                return;
+            }
+            if (randY < 2)
+            {
+                printf("Bail! y = %d < 2\n", randY);
+                return;
+            }
+
+            // Vanilla would use an `rcl` for the lower bit, getting the carry flag in.
+            // We're substituting it with an extra bit of the random state.
+            const bool topographyBit = randX > randY;
```

Comment:
This feels odd but im pretty sure its how vanilla did it. I think it would mean that hills are more common to one edge though.